src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -2707,6 +2707,12 @@ void Unit::_UpdateSpells(uint32 time)
         }
     }
 
+    for (auto nexctSpellCrit : m_spellCritMap)
+    {
+        if (!nexctSpellCrit.first.first)
```

Comment:
not yet, will do if i'm home.. wanted to test it. Pushed it first in the repo.. but thanks for your hint, i will try later